addons/mail/static/src/core/common/messaging_service.js

```diff
@@ -78,7 +78,8 @@ export class Messaging {
         this.store.discuss.starred.counter = data.starred_counter;
         this.store.mt_comment_id = data.mt_comment_id;
         this.store.discuss.isActive =
-            data.menu_id === this.router.current.hash?.menu_id ||
+            (data.menu_id === this.router.current.hash?.menu_id &&
+                this.router.current.hash.action !== "menu") ||
```

Comment:
Only action is reliable, `menu_id` is bad either way and we could ignore the world at this rate.

You can check by refreshing while being on the channel config page, it has the same `menu_id` as discuss app but yet another action, the bug is still present there and condition on `action` !== "menu" ` won't fix it)

`/web?debug=assets#model=discuss.channel&view_type=kanban&cids=1&menu_id=79`

Reply:
The issue is that we return `menu_id` rather than `action_id` in init messaging. This is flawed by design because we don't return the correct value.

Reply:
fix is `_xmlid_to_res_id("mail.menu_root_discuss")` -> `_xmlid_to_res_id("mail.action_discuss")` and then compare to action

Reply:
And in stable we should keep the support for both in theory, taking priority for `action_id` when given to avoid the bug.